api/repositories/app_repository.go

```diff
@@ -363,6 +368,17 @@ func (f *AppRepo) SetAppDesiredState(ctx context.Context, authInfo authorization
 	return cfAppToAppRecord(*cfApp), nil
 }
 
+func (f *AppRepo) DeleteApp(ctx context.Context, authInfo authorization.Info, message AppDeleteMessage) error {
+	cfApp := &workloadsv1alpha1.CFApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      message.AppGUID,
+			Namespace: message.SpaceGUID,
+		},
+	}
+
+	return f.privilegedClient.Delete(ctx, cfApp)
```

Comment:
Would it work still if we used the user client like we do in AppList?

https://github.com/cloudfoundry/cf-k8s-controllers/pull/360/files#diff-433641cf784e6cc94960970a9af8ae3a9c392337e21bc68d90485dd5c9a31b10R200

Reply:
@tcdowney It works with some minor RBAC tweaks.